sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py

```diff
@@ -36,13 +35,16 @@ class CloudEvent(EventMixin):   #pylint:disable=too-many-instance-attributes
     """Properties of an event published to an Event Grid topic using the CloudEvent 1.0 Schema.
 
     All required parameters must be populated in order to send to Azure.
+    If data is of binary type, data_base64 can be used alternatively. Note that data and data_base64
+    cannot be present at the same time.
```

Comment:
may I ask when the user should use `data_base64` and when should the user just use `data`?
it looks like `data` is enough to handle bytes.

Reply:
yes - data is enough to handle bytes - when user passes bytes using data, we internally send data_base64 instead - something like this
 ```
if isinstance(self.data, six.binary_type):
            data_base64 = self.data
            data = None
```

but, we should not stop users from sending data_base64 directly as a keyword instead of data

Reply:
is there any reason why we want to add `data_base64` argument to emphasize the `base64` encoded bytes?
I still feel that if we're being smart to handle all the input data type, then this `data_base64` seems redundant.

Reply:
we want a data_base64 since https://github.com/cloudevents/spec says so :)